mesonbuild/compilers.py

```diff
@@ -1053,7 +1053,7 @@ def compiles(self, code, env, extra_args=None, dependencies=None, mode='compile'
     def _links_wrapper(self, code, env, extra_args, dependencies):
         "Shares common code between self.links and self.run"
         args = self._get_compiler_check_args(env, extra_args, dependencies, mode='link')
-        return self.compile(code, args.to_native())
+        return self.compile(code, args)
```

Comment:
This will mean that `-L` flags will not get converted to `/LIBPATH` properly with MSVC. This is particularly a problem with user-provided `link_args`. If tests don't fail because of this, it's because the tests are incomplete, not because nothing will break.